lib/FrontendTool/FrontendTool.cpp

```diff
@@ -219,38 +239,79 @@ static bool emitLoadedModuleTraceIfNeeded(ASTContext &ctxt,
     return true;
   }
 
-  llvm::SmallVector<std::string, 16> swiftModules;
+  ModuleDecl* currentModule = ctxt.getModuleByName(moduleName);
+  ModuleDecl::ImportFilter filter = ModuleDecl::ImportFilterKind::Public;
+  filter |= ModuleDecl::ImportFilterKind::Private;
+  filter |= ModuleDecl::ImportFilterKind::ImplementationOnly;
+  llvm::SmallVector<ModuleDecl::ImportedModule, 8> imports;
+  currentModule->getImportedModules(imports, filter);
+
+  llvm::StringSet<> importedModulePaths;
+  for (std::pair<ModuleDecl::AccessPathTy, ModuleDecl *> &import : imports)
+    importedModulePaths.insert(import.second->getModuleFilename());
```

Comment:
This would probably be better to store the `ModuleDecl` (faster to look up later). You can use `llvm::SmallPtrSet` for that.